packages/react/src/publication/usePublication.ts

```diff
@@ -26,66 +49,81 @@ export type UsePublicationArgs = OneOf<PublicationRequest>;
  * });
  * ```
  *
+ * ## Basic Usage
+ *
+ * Get Publication by Id:
+ *
+ * ```ts
+ * const { data, error, loading } = usePublication({
+ *   forId: '0x04-0x0b',
+ * });
+ * ```
+ *
+ * Get Publication by Transaction Hash:
+ *
+ * ```ts
+ * const { data, error, loading } = usePublication({
+ *   forTxHash: '0xcd0655e8d1d131ebfc72fa5ebff6ed0430e6e39e729af1a81da3b6f33822a6ff',
+ * });
+ * ```
+ *
+ * ## Suspense Enabled
+ *
+ * You can enable suspense mode to suspend the component until the session data is available.
+ *
+ * ```ts
+ * const { data } = usePublication({
+ *   forId: '0x04-0x0b',
+ *   suspense: true,
+ * });
+ *
+ * console.log(data.id);
+ * ```
+ *
  * @category Publications
  * @group Hooks
+ *
  * @param args - {@link UsePublicationArgs}
  */
 export function usePublication({
   forId,
   forTxHash,
-}: UsePublicationArgs): ReadResult<AnyPublication, NotFoundError | UnspecifiedError> {
+}: UsePublicationArgs<never>): ReadResult<AnyPublication, NotFoundError | UnspecifiedError>;
+export function usePublication(
+  args: UsePublicationArgs<true>,
+): SuspenseResultWithError<AnyPublication, NotFoundError>;
+export function usePublication({
+  suspense = false,
+  ...request
+}: UsePublicationArgs<boolean>): UsePublicationResult {
   invariant(
-    forId === undefined || forTxHash === undefined,
+    request.forId === undefined || request.forTxHash === undefined,
     "Only one of 'forId' or 'forTxHash' should be provided to 'usePublication' hook",
   );
 
-  const { data, error, loading } = useReadResult(
-    usePublicationHook(
-      useLensApolloClient({
-        variables: useFragmentVariables({
-          request: {
-            ...(forId && { forId }),
-            ...(forTxHash && { forTxHash }),
-          },
-        }),
-        fetchPolicy: 'cache-and-network',
-        // leverage cache content if possible
-        nextFetchPolicy: 'cache-first',
-      }),
-    ),
-  );
-
-  if (loading) {
-    return {
-      data: undefined,
-      error: undefined,
-      loading: true,
-    };
-  }
+  const result = useSuspendableQuery<AnyPublication | null, PublicationVariables>({
+    suspense,
+    query: PublicationDocument,
+    options: useLensApolloClient({
+      variables: useFragmentVariables({ request }),
+      fetchPolicy: 'cache-and-network',
+      nextFetchPolicy: 'cache-first',
+    }),
+  });
 
-  if (error) {
-    return {
-      data: undefined,
-      error,
-      loading: false,
-    };
-  }
+  // if (request.forId && !isValidPublicationId(request.forId)) {
```

Comment:
there is a problem, that with wrong publicationId the API returns error, but it's not handled anywhere so it's thrown in the example app, would it make sense to do simple publicationId validation, the same as on the API side? the other thing is, it is not possible before useSuspendableQuery hook and later is too late as the API error is already thrown. any idea?

Reply:
or that use case shouldn't really exist? I made this UI and it breaks badly when using a wrong publication Id https://lens-sdk-example-web-wagmi-q0w3lfali-avaraxyz.vercel.app/publications/usePublication

Reply:
to me this is a case of wrong usage, a wrong Id (from a format perspective) to me should throw.

Reply:
I rebased my branch on top of this one. I polished this aspect there. Will merge both now.